app/plugins/ee-openid-connect.md

```diff
@@ -0,0 +1,130 @@
+This plugin can be used to implement Kong as a (proxying) [OAuth 2.0][oauth2] resource
+server (RS) and / or as a OpenID Connect relying party (RP) between the client
+and the upstream service.
+
+The plugin supports several types of credentials. Currently it can be used with
+the following types of credentials:
+
+- Signed [JWT][jwt] access tokens ([JWS][jws]) with the standardised signing algorithms ([JWA][jwa])
+- Opaque access tokens with either Kong OAuth 2.0 plugin issued tokes or 
+  3rd party IdP issued ones through token introspection (IdP needs to support it)
+- Username and password through the OAuth 2.0  password grant (the plugin will
+  automatically exchange such credentials with access token by calling IdP's token
+  endpoint)
+- Client id and secret through the OAuth 2.0 client credentials grant (the plugin
+  will automatically exchange such credentials with access token by calling IdP's
+  token endpoint)
+- Authorization code that plugin can retrieve from the client when using
+  OpenID Connect authorization code flow
+- Session cookie credentials that the plugin can setup between the client and Kong
+  (usually used with web browser clients together with authorization code grant)
+  
+In addition to that, the plugin can automatically refresh the access token using
+a refresh token, providing that the plugin has access to refresh token, and that
+can be a case when the plugin does the token exchange with proper scopes included,
+such as `offline_access`.
+
+You can either let the plugin to exclusively talk to your IdP as a trusted client
+(and let it do all the credential exchange) or you can let clients talk to IdP
+directly, and then present access token to the upstream service protected with
+the OpenID Connect plugin (or you can do both).
+
+Some of the capabilities of the plugin are listed below:
+
+- [WebFinger][webfinger] and [OpenID Connect Discovery][discovery]
+- [ID Token][idtoken] verification
+- [UserInfo][userinfo] endpoint data injecting
+- [RP-Initiated Logout][rplogout]
+- [OAuth 2.0 Token Revovation][revocation] during the logout (optionally)
+- [OAuth 2.0 Token Introspection][introspection] support
+- [OAuth 2.0 Proof Key for Code Exchange][pkce] (PKCE) support
+- Standard and configurable claims verification
+- Caching (optional) of token, introspection and user info endpoint request
+
+The plugin has been tested with several OpenID Connect capable providers such as:
+
+- [Auht0][auth0]
+- [Connect2id][connect2id]
+- [Dex][dex]
+- [Gluu][gluu]
+- [Google][google]
+- [IdentityServer4][identityserver4]
+- [Keycloak][keycloak]
+- [Microsoft Azure Active Directory v1][azurev1]
+- [Microsoft Azure Active Directory v2][azurev2]
+- Microsoft Live Connect
+- [Okta][okta]
+- [OneLogin][onelogin]
+- [OpenAM][openam]
+- [Paypal][paypal]
+- [PingFederate][pingfederate]
+- [Salesfoce][salesforce]
+- [Yahoo!][yahoo]
+
+As long as your provider supports OpenID Connect standards the plugin will
```

Comment:
apart from the slightly awkward way this sentence reads (maybe lacking some punctuation?), I this we should be less affirmative here, e.g.: "the plugin _should_ work"

Let's be more reserved about its capabilities and speculative support (just "best practices" we should abide by, and do already through-ought our description of other potentially supported dependencies , like untested OpenResty versions etc...)